### PR TITLE
gcp-o11y: Remove monitored resource detection for logging

### PR DESCRIPTION
**Pull request by DNVindhya**
b/277095016

This PR removes the monitored resource detection for logging (which is only available for kubernetes environments).

By not populating monitored resource, we will leverage cloud logging library's [auto population of metadata](https://github.com/googleapis/java-logging/blob/v3.14.5/README.md#auto-population-of-log-entrys-metadata) for monitored resource type and values.

### CODE REVIEW

File: gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java [hunk 1]
```diff
@@ -178,34 +167,14 @@ Logging createLoggingClient() {
   }
 
   @VisibleForTesting
-  static Map<String, String> getCustomTags(Map<String, String> customTags,
-      Map<String, String> locationTags, String projectId) {
+  static Map<String, String> getCustomTags(Map<String, String> customTags) {
```
Comment (sanjaypujare) [inline gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java:170]: This function effectively does very little/nothing - handle `null` on line 172 may be. Can you just inline this function?

**Review by ejona86 — APPROVED** [review]

File: gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java (hunk 1 above)
Comment (DNVindhya) [reply gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java:170]: The function also creates an immutable map of `customTags`. On further looking realized `null` check is not required, as [ObservabilityConfigImpl](https://github.com/grpc/grpc-java/blob/master/gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfigImpl.java#L120-L133) will always return `Collections.emptyMap()` when custom tags are not set in the configuration.

So I have two options:
1. Remove `null` check and keep the function as-is to create an immutable map.
2. Make the method inline and remove `null` check i.e
```
this.customTags = config.getCustomTags();
```

I prefer Option 1, as it gives us an immutable map. Lmk what you think.

Comment (sanjaypujare) [reply gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java:170]: Okay sounds good- option 1 is ok

**Review by sanjaypujare — APPROVED** [review]
LGTM

**Pull request merged**